src/web/screens/sharingPermissions.tsx

```diff
@@ -41,6 +42,7 @@ function SharingPermissions() {
         } saved to Your Sharing Permissions`,
       });
       setSharedTypes(response.allowed_types);
+      if (!participant?.completedRecommendations) await CompleteRecommendations(participant!.id);
```

Comment:
do we need to reload participant after this?

Reply:
I was initially thinking no, but I think it is probably a good idea. How would I trigger a reload given the participant is using `useContext`?
```
const { participant } = useContext(ParticipantContext);
```

Reply:
I think we could use loaded the updated participant and use setParticipant to set the latest participant

Reply:
ahh right, didn't realise we could just do the following to get the setter:
```
const { participant, setParticipant } = useContext(ParticipantContext);
```

Reply:
how's that look now?

Reply:
actually maybe it makes more sense for `CompleteRecommendations()` to just return the updated participant, hence I won't need the extra call to `GetCurrentUsersParticipant()`

Reply:
Yea, I think let CompleteRecommendations() return the updated participant, and then call setParticipant make more sense.

Reply:
I realised that we have a `useEffect` that has a dependency on participants. However, participant.id is optional to GetSharingList. Could we move this call into the router loader? So that the participant updates will not make this unnecessary call
```
  const loadSharingList = useCallback(async () => {
    const response = await GetSharingList(participant!.id);
    setSharedSiteIds(response.allowed_sites);
    setSharedTypes(response.allowed_types ?? []);
  }, [participant]);

  useEffect(() => {
    loadSharingList();
  }, [loadSharingList]);
```

Reply:
**tl;dr**: I think it is better to just remove the optional parameter on the `GetSharingList` call and remove the `participant` dependency on the callback.

I tried putting this into the router loader but here are my findings:
- Since <BulkAddPermissions>, <SearchAndAddParticipants> and <SharingPermissionsTable> all need the resolvedSharingList, I believe they have to be wrapped into one big <Await>, which means resolvedParticipantTypes has to be roped in too. This means that each time the useRevalidator is called, it also needs to run the api calls to participantTypes, available and sharingPermission again. Whereas the original solution does not need to make these three calls, as the updated participant comes back from the add endpoint anyway. If there is a way to avoid having all three components within the same <Await> then useLoaderData might seem like a more viable solution.
- Per the [docs](https://reactrouter.com/en/main/hooks/use-revalidator), useRevalidator should not really be used for normal CRUD operations on data in response to user interactions, instead, should probably use <Form>, useSubmit or useFetcher. I'm not sure if those alternatives are appropriate here, but thought it was an interesting point anyway.
- It could just be the way I have implemented it, but it also seems to cause a flicker in the UI where the components within the <Await> disappear then quickly reappear 
- I'm thinking I'll just merge in the above solution which is to remove the participant as a dependency for the useEffect , and just call GetSharingList() without the optional participantId parameter. I think if all the three components needed those two variables to be resolved every time, then the useLoaderData pattern may be appropriate, but given that is not the case, I think keeping the two local state variables and the useEffect is a better solution.